### PR TITLE
[NGS-589] Move OptSOA Behind Prebid

### PR DESCRIPTION
**Pull request by shahbaztariq**
## Context

...

## How To Test

...

## Related PRs

- TJX - https://github.com/Tapjoy/mediation_bid_service/pull/1087
- Prebid - https://github.com/Tapjoy/tpe_prebid_service/pull/99
- go-prebid - https://github.com/Tapjoy/go-prebid/pull/77

## Jira

https://tapjoy.atlassian.net/browse/NGS-589

### CODE REVIEW

**Review by umutdemirel — APPROVED** [review]
Thanks for the hard work here. 1 critical comment, but related to Chef. LGTM!

**Review by alanbrent — COMMENTED** [review]
It would be best if you could deploy the Chef PR first to set up all your environment variables, and then deploy the app changes. That way you don't have to remember to circle back and remove the hardcoded values.

This is my _preference_, so ultimately this is a non-blocking comment.

File: config/config.go [hunk 1]
```diff
@@ -969,6 +969,8 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetDefault("adapters.sovrn.endpoint", "http://ap.lijit.com/rtb/bid?src=prebid_server")
 	v.SetDefault("adapters.synacormedia.endpoint", "http://{{.Host}}.technoratimedia.com/openrtb/bids/{{.Host}}")
 	v.SetDefault("adapters.tappx.endpoint", "http://{{.Host}}")
+	v.SetDefault("adapters.tapjoy.disabled", true)
+	v.SetDefault("adapters.tapjoy.endpoint", "http://legacy-discovery:7286/bid")
```
[the remark this answers is not in the document]
Comment (umutdemirel) [reply config/config.go:973]: This is a great call, a heads up actually :) Thanks Brent.

When I traced back, I'm seeing that this is correct but you need to update the Chef one. That env variable should have been `PBS_ADAPTERS_TAPJOY_ENDPOINT`. Viper config in this prebid project parses this ENV and put into `"adapters.tapjoy.endpoint"`. This default value actually guards the wrong ENV but to be consistent let's fix Chef with the correct one for future use.

BTW, XAPI comes into play when we have more than 1 endpoint for a DSP.

**Review by umutdemirel — APPROVED** [review]
LGTM!

**Pull request merged**